pkg/chunk/cache/memcached.go

```diff
@@ -239,11 +250,15 @@ func (c *Memcached) Store(ctx context.Context, keys []string, bufs [][]byte) {
 
 // Stop does nothing.
 func (c *Memcached) Stop() {
-	if c.inputCh == nil {
+	if c.quit == nil {
 		return
 	}
 
-	close(c.inputCh)
+	select {
+	case <-c.quit:
+	default:
+		close(c.quit)
```

Comment:
Can you explain this bit please?

Reply:
I just did a quick search and it looks like the ChunkFetcher has the same issue that in its Stop func closes the same channel that it is sending on https://github.com/cortexproject/cortex/blob/master/pkg/chunk/chunk_store_utils.go#L120.

The reason for the select case is to check if the quit channel has already been closed or not, so that multiple calls to the Stop func don't result in a panic for trying to close a closed channel.

Reply:
OK I was actually missing the fact that cases with a communication take priority over `default` case. [Go spec](https://golang.org/ref/spec#Select_statements).
I had it in my mind that Go `select` statements do not prioritise one case over another.

Reply:
do you want me to update the ChunkFetcher as well in this PR?

Reply:
Sure, if you think it will crash in the same way.

Reply:
it has the same potential to crash, but I don't know what and when that code is used so unsure how likely it is to occur there, I have updated the PR to include a fix for it just in case